### PR TITLE
Update learning rate on each backward pass instead of each forward pass.

### LINKED ISSUE #1476: Learning rate scheduler should step after each optimizer step
## 🐛 Bug

I'm not sure that this is a bug or if it is a deliberate design decision, but right now the learning rate schedule gets updated at every "step" which actually corresponds to every forward pass. I think a more standard implementation would have the learning rate scheduler "step" interval correspond to being updated every backwards pass. This has caused me a lot of problems with instability as I did not realize that using standard learning rate warmups of say 16000 steps would actually only warm up for 1000 steps if I set `accumulate_grad_batches=16`.

### PR DESCRIPTION
**Pull request by rmrao**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1476.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rmrao** [comment]
I believe at the moment that `run_training_batch` runs a forward pass, but only runs a backwards pass if `self.batch_idx + 1 % self.accumulate_grac_batches == 0` (see [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/f293c9b5f4b4f9fabb2eec0c369f08a66c57ef14/pytorch_lightning/trainer/training_loop.py#L624-L648)). This is consistent with how the global step is incremented in `run_training_epoch` ([here](https://github.com/PyTorchLightning/pytorch-lightning/blob/f293c9b5f4b4f9fabb2eec0c369f08a66c57ef14/pytorch_lightning/trainer/training_loop.py#L493-L494)). 

I didn't move the learning rate update line - just added a condition over it. It might be better / cleaner to move it somewhere else, or to try and move it inside of `run_training_batch`. Doing that would require rewriting how `self.update_learning_rates` works, however.

**Comment by williamFalcon** [comment]
ok got it. that makes sense. it does seem a little dirty at the moment.
Mind adding a todo comment on top of your condition so we know to look into this later?

**Comment by rmrao** [comment]
Done - added a TODO to potentially merge optimizer step, lr update, and global step increment.
